libraries/standard/mqtt/cbmc/proofs/MQTT_Init/MQTT_Init_harness.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file MQTT_Init_harness.c
+ * @brief Implements the proof harness for MQTT_Init function.
+ */
+
+#include "mqtt.h"
+
+void harness()
+{
+    MQTTContext_t context;
+    MQTTTransportInterface_t transportInterface;
+    MQTTApplicationCallbacks_t callbacks;
+    MQTTFixedBuffer_t networkBuffer;
+
+    MQTT_Init( nondet_bool() ? NULL : &context,
+               nondet_bool() ? NULL : &transportInterface,
+               nondet_bool() ? NULL : &callbacks,
+               nondet_bool() ? NULL : &networkBuffer );
+}
```

Comment:
This is an example of why you might want allocators for these things.  An allocator would call mallocCanFail which could return either null or a valid pointer.  As you write more proofs, you may want to revisit this.